src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/OcrValidator.java

```diff
@@ -0,0 +1,105 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.validation;
+
+import io.vavr.control.Try;
+import joptsimple.internal.Strings;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.data.util.Optionals;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator;
+import uk.gov.hmcts.reform.bulkscanprocessor.config.ContainerMappings;
+import uk.gov.hmcts.reform.bulkscanprocessor.exceptions.OcrValidationException;
+import uk.gov.hmcts.reform.bulkscanprocessor.model.blob.InputEnvelope;
+import uk.gov.hmcts.reform.bulkscanprocessor.model.blob.InputScannableItem;
+import uk.gov.hmcts.reform.bulkscanprocessor.ocrvalidation.client.OcrValidationClient;
+import uk.gov.hmcts.reform.bulkscanprocessor.ocrvalidation.client.model.req.FormData;
+import uk.gov.hmcts.reform.bulkscanprocessor.ocrvalidation.client.model.req.OcrDataField;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import static java.util.stream.Collectors.toList;
+
+@Component
+@EnableConfigurationProperties(ContainerMappings.class)
+public class OcrValidator {
+
+    private static final Logger log = LoggerFactory.getLogger(OcrValidator.class);
+
+    private final OcrValidationClient client;
+    private final ContainerMappings containerMappings;
+    private final AuthTokenGenerator authTokenGenerator;
+
+    //region constructor
+    public OcrValidator(
+        OcrValidationClient client,
+        ContainerMappings containerMappings,
+        AuthTokenGenerator authTokenGenerator
+    ) {
+        this.client = client;
+        this.containerMappings = containerMappings;
```

Comment:
Only `getMappings` used in one function. shall we just build object with that value instead? it'll be private anyway

Reply:
Not sure.
We'd still need a map (from poBox to Url).